app/views/contexts/summary/group.py

```diff
@@ -88,6 +88,26 @@ def _build_blocks_and_links(
         blocks = []
 
         for block in group_schema["blocks"]:
+            if block["type"] == "ListRepeatingQuestion":
+                # list repeating questions aren't themselves on the path, the parent list collector is checked in ListCollectorBlock
+                list_collector_block = ListCollectorBlock(
```

Comment:
> group.py now checks if the block being rendered is a repeating question. It needs to do this before checking if the block is on the path, as list repeating questions won't be on it, its a property of the list collector instead. This will call a new method on ListCollectorBlock to return a list of blocks for the question, one copy for each list item id

I'm not following why we are still doing this. Is this logic calc summary specific?. We handle list collector already which should cover repeating block answers, why do we need this here now?.

This adds repeating blocks even if the list collector isn't for it isn't on the path.

Reply:
Yep - its calc summary specific at the moment, we handle list collectors, but we don't handle list repeating questions, and a calculated summary won't necessarily use all the repeating blocks and answers, so I can't use the existing "ListCollector" case as this is rendering summary elements including all questions rather than the specific ones that have been passed to group in the calc summary case

Reply:
Will fix the issue with list collectors that arent on the path

Reply:
Same here, not sure if I fully follow. Isn't this going to add repeating blocks to the list of blocks for all summaries?. It should only be when it is a calculated summary right?.

Reply:
Added a comment as per our chat to say that this relates to calculated summary types and not section/final summary types

Reply:
The paths are still not checked?. It is still adding the blocks regardless?.

Reply:
Fixed 👍